docs/multiregion.md

```diff
@@ -0,0 +1,98 @@
+* Legalese & policy --> for some legal requirements/policy certain things need be separate.
+
+## Proposed solution -- introducing region concept
+
+To solve the above mentioned issues, we'll introduce managed kubecarrier federation. The basic concepts would be:
+
+* ServiceCluster --> stays the same as before; its akin to availability zone.
+* Region --> a group of multiple providers & service clusters regionally grouped with reasonable low-latency network connection is a region. The region has a single KubeCarrier management cluster
+
+This would provide the logical, geographical, and failure domain grouping to acchive the initial motivational goals. How to coordinate and configure the global KubeCarrier solution from single place (or to the end user appearing as a single place) is still an open problem.
+
+Between regions the following things should be synced, and should be globally unique:
+* `(ServiceCluster, Account)`
+
+Additionally the following objects should be replicated across every region:
+* `(Account)`
+
+Thus there must exist global consensus what the global KubeCarrier configuration should be. The authorization shall be handled the same as in the single region case --> management cluster shall create the necessary roles and rolebinding for the account.
+
+This is enough to handle the authorization via the native k8s mechanism. For Authentication see the dedicated section.
+
+## related work:
+
+* https://github.com/kubermatic/kubecarrier/pull/291/files
+
+This solutions revolves around having a single superCluster managing multiple regions with, and each region being managed by region-local installation.
+
+Pros:
+* the user authentication happens at the superCluster level, whilst their actions is propagated to region-local management cluster via the `Impersonate` headers
+* installation is handled via the superCluster operator
+
+Cons:
+* the latency issue still remains. (e.g. request for creating new service instance would propagate from user -- superClusterManager -- region -- region local serviceCluster)
+* the superCluster becomes Single point of Failure
+* the superCluster has admin privileges in the region's management cluster. A exploit in our API server implementation could lead to unwanted privilege escalation.
+
+## Proposed solution:
+
+### Having no superCluster/Federations
+
+* The latency and SPOF issue shall still reamain.
+* We build API server accepting user's kubeconfig/token
+    * This way we're preventing privilege escalation since API server doesen't have cluster-admin privileges.
+* it's simpler approach
+
+### build truly decentralized KubeCarrier installation with helper API/library for cross-region management
+
+The way cross-region requires consists of two operation types:
+* routing
+* fan-out
+
+Also there are two ways for keeping configuration in sync:
+* static configuration
+* central configuration plane (akin to superCluster in the original @nico proposal)
```

Comment:
I am a bit lost here, does this mean that we still need to have a central kubecarrier installation? If yes, how this is truly decentralized?

Reply:
You have to somehow handle the region discovery. There are a couple of solutions here, from having it hardcoded in the clients, to having a centralized configuration plane, all the way to DNSesque or BitTorrent peer-discovery decentralized region discovery. Here I'm listing a few options.